src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -220,6 +220,21 @@ ApplicationSharedPtr FindApp(DataAccessor<ApplicationSet> accessor,
 }
 
 template <class UnaryPredicate>
+ApplicationSharedPtr FindApp(
+    DataAccessor<ApplicationManagerImpl::AppsWaitRegistrationSet> accessor,
+    UnaryPredicate finder) {
+  ApplicationManagerImpl::AppsWaitRegistrationSet::iterator it = std::find_if(
```

Comment:
@AGritsevich so in case we have two functions which are search app could the be joined?
I mean if bad code present somewhere in project the best practice is to fix it and not copy&paste